lib/flipr.js

```diff
@@ -1,93 +1,63 @@
-'use strict';
-
-var async = require('async');
-var _ = require('lodash');
-var configReader = require('./config-reader/config-reader');
-var connectReqConfig = require('./middleware/connect-req-config');
-var idToPercent = require('./util/id-to-percent');
-var continueIfValid = require('./util/continue-if-valid');
-var validateKey = require('./validate/validate-key');
-var defaultInputValidator = require('./validate/default-input-validator');
-var validateRuleSync = require('./validate/validate-rule-sync');
-var validateOptions = require('./validate/validate-options');
-var validateRule = require('./validate/validate-rule');
-
-module.exports = flipr;
-
-var defaultOptions = {
-  inputValidator: defaultInputValidator,
-  rules: []
-};
-
-function flipr(input, key, cb) {
-  //makes input and key optional
-  if(_.isFunction(input)) {
-    cb = input;
-    input = null;
-  } else if (_.isFunction(key)) {
-    cb = key;
-    key = null;
+const validateRule = require('./rules/validate-rule');
+const getValueByRules = require('./rules/get-value-by-rules');
+const idToPercent = require('./id-to-percent');
+
+const defaultOptions = { rules: [] };
+
+class Flipr {
+  constructor(options) {
+    this.options = {
+      ...defaultOptions,
+      ...options,
+    };
+    if (!this.options.source) {
+      throw new Error('options.source is required');
+    }
+    this.source = this.options.source;
+    this.options.rules.map(validateRule);
   }
 
-  if(!_.isFunction(cb))
-    throw new Error('flipr: cb must be a function');
-
-  if(input === null)
-    flipr.getDictionary(cb);
-  else if(key === null)
-    flipr.getDictionaryByRules(input, cb);
-  else
-    flipr.getValueByRules(input, key, cb);
-}
-
-flipr.cachedOptions = defaultOptions;
-
-flipr.init = function init(options) {
-  validateOptions(options);
-  options = _.defaults(options, defaultOptions);
-  _.map(options.rules, validateRuleSync); //This will throw if rules are invalid
-  flipr.cachedOptions = options;
-  configReader.init(options);
-  return connectReqConfig;
-};
-
-flipr.getDictionary = configReader.getDictionary;
+  _getValue(config, key, input) {
+    // value (static) always overrides values (dynamic) on config item
```

Comment:
Should probably mention this in readme in the `Other Noteworthy Behavior` section